### PR TITLE
Use regular agent for triggering e2e tests

### PR DESCRIPTION
**Pull request by SomeoneToIgnore**
e2e triggering is a bunch of `curl` calls on GH API + a few secrets, so could be done from any agent, not a special rust one.

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -410,10 +410,7 @@ jobs:
         shell: bash -euxo pipefail {0}
 
   trigger-e2e-tests:
-    runs-on: dev
-    container:
-      image: 369495373322.dkr.ecr.eu-central-1.amazonaws.com/rust:pinned
-      options: --init
+    runs-on: 'ubuntu-latest'
```
Comment (LizardWizzard) [inline .github/workflows/build_and_test.yml:413]: There is `369495373322.dkr.ecr.eu-central-1.amazonaws.com/base:latest` image which might be preferable if the current policy is "use our runners everywhere except for mac os compile job". @zoete wdyt?

Comment (zoete) [reply .github/workflows/build_and_test.yml:413]: The rust one might be a bit overkill for this purpose, the most recent `base:pinned` image has curl

Comment (SomeoneToIgnore) [reply .github/workflows/build_and_test.yml:413]: Is there such a thing as `base:pinned`? Have not found that, so replaced with `base:latest`, as suggested.

Comment (zoete) [reply .github/workflows/build_and_test.yml:413]: Yes there is as you can see here: https://eu-central-1.console.aws.amazon.com/ecr/repositories/private/369495373322/base?region=eu-central-1

**Review by bayandin — APPROVED** [review]
Nice!

**Pull request merged**